login.js

```diff
@@ -1,28 +1,67 @@
 import storage from './storage.js';
-import { storageHost } from './constants.js'
+import {getContractSource, hexToWordList, wordListToHex} from './blockchain.js';
+import {storageHost} from './constants.js'
 
 const loginEndpoint = 'https://login.exokit.org';
-const usersEndpoint = 'https://users.exokit.org';
+// const usersEndpoint = 'https://users.exokit.org';
 
 const _clone = o => JSON.parse(JSON.stringify(o));
 
 let loginToken = null;
 let userObject = null;
-async function pullUserObject() {
-  const res = await fetch(`${usersEndpoint}/${loginToken.name}`);
-  if (res.ok) {
-    userObject = await res.json();
-  } else if (res.status === 404) {
-    userObject = {
-      name: loginToken.name,
-      avatarHash: null,
-      inventory: [],
-    };
-  } else {
-    throw new Error(`invalid status code: ${res.status}`);
+async function ensureUserObjectBaked() {
+  const contractSource = await getContractSource('isUserAccountBaked.cdc');
+
+  const res = await fetch(`https://accounts.exokit.org/sendTransaction`, {
+    method: 'POST',
+    body: JSON.stringify({
+      limit: 100,
+      script: contractSource.replace(/ARG0/g, '0x' + loginToken.addr),
+      wait: true,
+    }),
+  });
+  const response = await res.json();
+  const isBaked = response.encodedData.value;
+  if (!isBaked) {
+    const contractSources = await getContractSource('bakeUserAccount.json');
+    for (const contractSource of contractSources) {
+      contractSource.address = loginToken.addr;
+      contractSource.mnemonic = loginToken.mnemonic;
+      contractSource.limit = 100;
+      contractSource.wait = true;
+
+      const res = await fetch(`https://accounts.exokit.org/sendTransaction`, {
+        method: 'POST',
+        body: JSON.stringify(contractSource),
+      });
+      
+      const response = await res.json();
+      console.log('baked account result', response);
+    }
   }
 }
-async function pushUserObject() {
+async function pullUserObject() {
+  await ensureUserObjectBaked();
+  
+  const contractSource = await getContractSource('getUserData.cdc');
+
+  const res = await fetch(`https://accounts.exokit.org/sendTransaction`, {
+    method: 'POST',
+    body: JSON.stringify({
+      limit: 100,
+      script: contractSource.replace(/ARG0/g, '0x' + loginToken.addr),
+      wait: true,
+    }),
+  });
+  const response = await res.json();
+  const name = response.encodedData.value[0].value ? response.encodedData.value[0].value.value : 'Anonymous';
+  const avatarHash = response.encodedData.value[1].value && response.encodedData.value[1].value.value;
```

Comment:
value[1].value.value pretty weird

Reply:
Yeah. Unfortunately I don't know a better way to peel the data off, though we might be able to abstract these more in the future.